setup.py

```diff
@@ -22,39 +22,37 @@
 import os
 import sys
 
-from setuptools import setup, find_packages, Extension, Feature
 from distutils.command.build_ext import build_ext
 from distutils.errors import CCompilerError
 from distutils.errors import DistutilsExecError
 from distutils.errors import DistutilsPlatformError
 import platform
 
+from setuptools import setup, find_packages, Extension
+
 py_impl = getattr(platform, 'python_implementation', lambda: None)
 is_pypy = py_impl() == 'PyPy'
 is_jython = 'java' in sys.platform
 
 codeoptimization_c = os.path.join('src', 'zope', 'i18nmessageid',
                                   "_zope_i18nmessageid_message.c")
-codeoptimization = Feature(
-    "Optional code optimizations",
-    standard=True,
-    ext_modules=[Extension(
+codeoptimization = [
+    Extension(
         "zope.i18nmessageid._zope_i18nmessageid_message",
         [os.path.normcase(codeoptimization_c)]
-        )])
+    ),
+]
 
 extra = {
-    'extras_require': {
-        'testing': ['nose', 'coverage'],
-        'docs': ['Sphinx'],
-    },
+
 }
 
+ext_modules = []
 if not is_pypy and not is_jython:
```

Comment:
Maybe please add an `and not is_pure` check with the os.getenv('PURE_PYTHON') logic copied from zope.interface etc.? So people can ask to skip extension modules.

Reply:
You know what, you're right and I was wrong.  Build-time checking of PURE_PYTHON will poison pip's wheel cache and we should fix zope.interface not to do that.

Reply:
I was slightly surprised to discover that you get different wheel file names depending on whether PURE_PYTHON is set (`zope.interface-4.7.2.dev0-py3-none-any.whl`) or not (`zope.interface-4.7.2.dev0-cp38-cp38-macosx_10_15_x86_64.whl`). I suspect it still messes with the cache though.